src/v/kafka/protocol/schemata/generator.py

```diff
@@ -1142,34 +1150,30 @@ class response;
 {%- endmacro %}
 
 {% macro conditional_tag_encode(tdef, vec) %}
-{%- if tdef.is_array %}
 {%- if tdef.nullable() %}
-{%- call tag_version_guard(tdef) %}
 if ({{ tdef.name }}) {
     {{ vec }}.push_back({{ tdef.tag() }});
 }
-{%- endcall %}
-{%- else %}
-{%- call tag_version_guard(tdef) %}
+{%- elif tdef.is_array %}
 if (!{{ tdef.name }}.empty()) {
     {{ vec }}.push_back({{ tdef.tag() }});
 }
-{%- endcall %}
-{%- endif %}
 {%- elif tdef.default_value() != "" %}
-{%- call tag_version_guard(tdef) %}
 if ({{ tdef.name }} != {{ tdef.default_value() }}) {
     {{ vec }}.push_back({{ tdef.tag() }});
 }
-{%- endcall %}
+{%- else %}
+{{ vec }}.push_back({{ tdef.tag() }});
```

Comment:
@graphcareful don't you want to gen `vec.reserve( {{tdefs.array | size}} )` or whatever

Reply:
question, all of these push_backs are surronded by conditionals, so it may be the case that we reserve space for n tags but then never push back a single one.

Reply:
I meant they are C++ runtime conditionals. So we can only know the maximum number of push backs (tdefs.array.size), not the actual number that occurs at runtime.

Reply:
right, but i'm saying that runtime check is basically free. 

```
if empty {
  // do the simple thing
  return early;
}

vec.reserve ( something )
for ( ...; ) 
   pushback( tdef.tag() );

return vec; 

```

Reply:
we may also reserve larger and then shrink to fit

Reply:
I whipped up some code that deletes the use of this vector all together, but i think that should be in a separate PR anyway, its sort of outside the scope of the small fix commits to tag handling here. 

To address this comment for now I will take michals advice and use reserve/shrink_to_fit